### ISSUE TITLE
one node produced an error: Error in ED2(data)

### ISSUE THREAD

**Issue by eshraaqsaeed** [hemberg-lab/SC3#103]
On running sc3 function, this error appears:
Error in checkForRemoteErrors(val) : 
  one node produced an error: Error in ED2(data) : 
  Not compatible with requested type: [type=list; target=double].

Code: (sce object on Yan dataset)
sce <- sc3(sce, ks = 6, biology = TRUE)

Session Info:
R version 4.0.4 (2021-02-15)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 8.1 x64 (build 9600)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods  
[9] base     

other attached packages:
 [1] scater_1.16.2               ggplot2_3.3.2               SC3_1.18.0                 
 [4] SingleCellExperiment_1.10.1 SummarizedExperiment_1.18.2 DelayedArray_0.14.1        
 [7] matrixStats_0.56.0          Biobase_2.48.0              GenomicRanges_1.40.0       
[10] GenomeInfoDb_1.24.2         IRanges_2.22.2              S4Vectors_0.26.1           
[13] BiocGenerics_0.34.0        

loaded via a namespace (and not attached):
 [1] viridis_0.5.1             BiocSingular_1.4.0        viridisLite_0.3.0        
 [4] foreach_1.5.0             DelayedMatrixStats_1.10.1 shiny_1.5.0              
 [7] doRNG_1.8.2               GenomeInfoDbData_1.2.3    vipor_0.4.5              
[10] robustbase_0.93-6         pillar_1.4.7              lattice_0.20-41          
[13] glue_1.4.2                digest_0.6.27             RColorBrewer_1.1-2       
[16] promises_1.1.1            XVector_0.28.0            colorspace_1.4-1         
[19] cowplot_1.0.0             htmltools_0.5.1.1         httpuv_1.5.4             
[22] Matrix_1.3-2              pcaPP_1.9-73              WriteXLS_6.2.0           
[25] pkgconfig_2.0.3           pheatmap_1.0.12           zlibbioc_1.34.0          
[28] purrr_0.3.4               xtable_1.8-4              mvtnorm_1.1-1            
[31] scales_1.1.1              later_1.1.0.1             BiocParallel_1.22.0      
[34] tibble_3.0.3              farver_2.0.3              generics_0.0.2           
[37] ellipsis_0.3.1            withr_2.3.0               ROCR_1.0-11              
[40] magrittr_2.0.1            crayon_1.3.4              mime_0.9                 
[43] doParallel_1.0.15         class_7.3-18              beeswarm_0.2.3           
[46] tools_4.0.4               lifecycle_0.2.0           munsell_0.5.0            
[49] cluster_2.1.0             rngtools_1.5              irlba_2.3.3              
[52] compiler_4.0.4            e1071_1.7-4               rsvd_1.0.3               
[55] tinytex_0.25              rlang_0.4.10              grid_4.0.4               
[58] RCurl_1.98-1.2            iterators_1.0.12          BiocNeighbors_1.6.0      
[61] rstudioapi_0.13           labeling_0.4.2            bitops_1.0-6             
[64] gtable_0.3.0              codetools_0.2-18          rrcov_1.5-5              
[67] R6_2.5.0                  gridExtra_2.3             dplyr_1.0.1              
[70] fastmap_1.0.1             ggbeeswarm_0.6.0          Rcpp_1.0.6               
[73] vctrs_0.3.2               DEoptimR_1.0-8            tidyselect_1.1.0         
[76] xfun_0.16    

**Additional Comment:
I restarted a new R session and the problem still existed.

Thank you in advance

**Comment by boyiguo1** [comment]
This previous issue seems relevant: https://github.com/hemberg-lab/SC3/issues/53#issuecomment-355736905
TLDR: you may want to see what is the data structure that contains your count/log count data. They should be `matrix`.
